src/function/gds/gds_utils.cpp

```diff
@@ -38,30 +38,37 @@ void GDSUtils::runFrontiersUntilConvergence(processor::ExecutionContext* context
     RJCompState& rjCompState, graph::Graph* graph, ExtendDirection extendDirection,
     uint64_t maxIters) {
     auto frontierPair = rjCompState.frontierPair.get();
+    auto outputNodeMask = rjCompState.outputWriter->getOutputNodeMask();
+    common::offset_t numReachedNodes = 0;
```

Comment:
The numReachedNodes is a shortest-paths-specific auxiliary for terminating early. It should be defined as part of the SPAlgorithms' shared state. And the SharedState or something similar should be passed to terminateFunction.